### ISSUE TITLE
Bad credentials for bucket when not running with --foreground

### ISSUE THREAD

**Issue by bkanuka** [GoogleCloudPlatform/gcsfuse#957]
I think this is a bit strange: gcsfuse only works when using the `--foreground` flag, otherwise I get a `Bad credentials for bucket __ permission denied` 

In the below `ce-vertex` is the name of the project and the bucket.

This does not work:
```
jupyter@vm-72d189e7-614c-42b0-a565-4bc40ada442a:~$ gcsfuse ce-vertex ce-vertex
2023/02/23 16:26:23.209457 Start gcsfuse/0.41.12 (Go version go1.18.4) for app "" using mount point: /home/jupyter/ce-vertex
2023/02/23 16:26:23.229709 Opening GCS connection...
daemonize.Run: readFromProcess: sub-process: mountWithArgs: mountWithConn: fs.NewServer: create file system: SetUpBucket: OpenBucket: Bad credentials for bucket "ce-vertex": permission denied
```

But this does:
```
jupyter@vm-72d189e7-614c-42b0-a565-4bc40ada442a:~$ gcsfuse --foreground ce-vertex ce-vertex
2023/02/23 16:26:32.941696 Start gcsfuse/0.41.12 (Go version go1.18.4) for app "" using mount point: /home/jupyter/ce-vertex
2023/02/23 16:26:32.941774 Opening GCS connection...
2023/02/23 16:26:32.942987 Creating a mount at "/home/jupyter/ce-vertex"
2023/02/23 16:26:32.943460 Creating a new server...
2023/02/23 16:26:32.943484 Set up root directory for bucket ce-vertex
2023/02/23 16:26:32.943491 OpenBucket("ce-vertex", "")
2023/02/23 16:26:33.039243 Mounting file system "ce-vertex"...
2023/02/23 16:26:33.042409 File system has been successfully mounted.
```

**Additional context**
This is running in a "Managed" Vertex Notebook on GCP.  The is a ui button built into Vertex to allow mounting a GCS bucket, and using this gives the same permission denied error.  Running gcsfuse from the terminal (in vertex) *only works with the foreground flag*

In case this is an issue with Vertex (I don't think it is) I opened a bug here:  https://issuetracker.google.com/issues/270545485

**Comment by vadlakondaswetha** [comment]
can you add --debug_fuse, --debug_gcs, --debug_http, --log-file  flags and share the logs.

**Comment by bkanuka** [comment]
Hmm so just to add, it looks like `gcsfuse` (without `--foreground`) is first attempting to auth using a token the VM metadata server instead of the `gcloud` configured Application Default Credentials.  I guess it's a matter of opinion if `gcsfuse` should always be trying to auth using `gcloud` before trying a token from the metadata server, so I'm not sure if this is a bug.

**Comment by vadlakondaswetha** [comment]
Vertex ai is configured to use VM metadata server instead of gcloud. When you run in --foreground mode which credentials is it using?

**Comment by bkanuka** [comment]
Without `--foreground` gcsfuse uses the VM metadata server (which in Vertex, returns some hidden/internal SA that I ),  but *with* `--foreground` gcsfuse uses gcloud, which is preconfigured to use the SA I own and configured Vertex to use.

So 2 little issues here:

- VM Metadata server is returning different credentials, not the SA that I've configured it to use (obviously not a `gcsfuse` issue)
- gcsfuse uses a different auth order with and without `--forground`.  Arguably this is a gcsfuse issue

**Comment by vadlakondaswetha** [comment]
I am able to reproduce to reproduce the issue and looks like the issue with vertex ai. GCSFuse doesn't anything special when --foreground flag is enabled. I am following up with vertex ai team and will keep this thread updated.

**Comment by bkanuka** [comment]
The bug I opened with Vertex was marked Infeasible (lol).   Hopefully you have more luck!  https://issuetracker.google.com/issues/270545485

**Comment by tarrade** [comment]
I was having similar issue with Vertex AI Notebook running with custom container. I could see gcs bucket in the same project but to see gcs bucket in other project it was only working with --forground.

Using gcsfuse 1.0.0 is solving my problem

**Comment by ashmeenkaur** [comment]
Thanks for the update, @tarrade! We appreciate you confirming that the issue has been resolved with gcsfuse 1.0.0.

@bkanuka, appreciate you bringing this issue to our attention. We attempted to reproduce the issue again using the latest Managed Vertex Notebook with GCSFuse 1.2.0 installed, but were unable to do so. This suggests that the issue has been addressed in GCSFuse 1.0.0.

If you encounter this issue again, please don't hesitate to reopen the issue.

Thank you for helping us improve GCSFuse!

Sincerely,
Ashmeen
